Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,43 @@
+import { Parser } from 'expr-eval';
 import React from 'react';
+import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
+import CalculatorControls from './components/DisplayComponents/CalculatorControls';
+
 import './App.css';
 
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { total: '' };
+  }
+
+  updateTotal = (text) => {
+    switch (text) {
+      case 'clear':
+        this.setState({ total: '' });
+        break;
+      case '=':
+        this.setState({ total: Parser.evaluate(
+          this.state.total
+            .replace('×', '*')
+            .replace('÷', '/')
+            .replace('–', '-')
+        )});
+        break;
+      default:
+        this.setState({ total: this.state.total + text });
+        break;
+    }
+  }
+
+  render() {
+    return (
+      <div className='app-container'>
+        <CalculatorDisplay text={this.state.total} />
+        <CalculatorControls updateTotal={this.updateTotal} />
```

Comment:
I like that you took control of the component structure and expanded upon it, it makes a lot more sense for the buttons to be separated from App.js and nested inside this CalculatorControls component you made